Failed tests (11667): TestRaftAfterRemoveRange

The following test appears to have failed:

[#11667](https://circleci.com/gh/cockroachdb/cockroach/11667):

```
I0120 05:36:36.980736 681 stopper.go:264  draining; tasks left:
2      storage/replica.go:1754
W0120 05:36:41.832697 681 storage/replica.go:1751  could not GC completed transaction: context deadline exceeded
I0120 05:36:41.832752 681 stopper.go:264  draining; tasks left:
1      storage/replica.go:1754
--- FAIL: TestRaftAfterRemoveRange (30.27s)
panic: timed out during shutdown [recovered]
	panic: timed out during shutdown [recovered]
	panic: timed out during shutdown

goroutine 7610 [running]:
testing.tRunner.func1(0xc8205a39e0)
	/usr/local/go/src/testing/testing.go:450 +0x171
github.com/cockroachdb/cockroach/util/leaktest.AfterTest(0x7f07b5a75d18, 0xc8205a39e0)
	/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:115 +0x194
github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).Stop(0xc8205d0140)
	/go/src/github.com/cockroachdb/cockroach/storage/client_test.go:288 +0x29d
github.com/cockroachdb/cockroach/storage_test.TestRaftAfterRemoveRange(0xc8205a39e0)
	/go/src/github.com/cockroachdb/cockroach/storage/client_raft_test.go:1092 +0x842
testing.tRunner(0xc8205a39e0, 0x1f42258)
	/usr/local/go/src/testing/testing.go:456 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:561 +0x86d

goroutine 1 [chan receive]:
testing.RunTests(0x187f838, 0x1f41040, 0xe0, 0xe0, 0x1)
	/usr/local/go/src/testing/testing.go:562 +0x8ad
testing.(*M).Run(0xc82078af08, 0x1)
	/usr/local/go/src/testing/testing.go:494 +0x70
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc82078af08)
	/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:37 +0x21
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc82078af08)
	/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:36 +0x26
main.main()
	github.com/cockroachdb/cockroach/storage/_test/_testmain.go:508 +0x113

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x2286fc0)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1029 +0x64
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0xbc

goroutine 7847 [semacquire]:
sync.runtime_Semacquire(0xc82089af3c)
	/usr/local/go/src/runtime/sema.go:43 +0x26
sync.(*WaitGroup).Wait(0xc82089af30)
	/usr/local/go/src/sync/waitgroup.go:126 +0xb4
github.com/cockroachdb/cockroach/storage.(*Replica).beginCmds(0xc8205b7380, 0xc820357970, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:738 +0x4fb
github.com/cockroachdb/cockroach/storage.(*Replica).addWriteCmd(0xc8205b7380, 0x7f07ba30b5e0, 0xc820115140, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:899 +0x100
github.com/cockroachdb/cockroach/storage.(*Replica).handleSkippedIntents.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1750 +0xdae
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask.func1(0xc82034a930, 0xc820946b20, 0x17, 0xc8202e07d0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:131 +0x58
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:132 +0x221

goroutine 7697 [chan receive]:
github.com/cockroachdb/cockroach/rpc.(*Server).sendResponses(0xc82062ad20, 0x7f07ba31ac58, 0xc8200d1980, 0xc82095a960)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:300 +0xa4
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func1(0xc82062ad20, 0x7f07ba31ac58, 0xc8200d1980, 0xc82095a960)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:207 +0x3f
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:208 +0x89c

goroutine 7671 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034aa10, 0xc820745740)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7663 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc820297400)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x490

goroutine 7676 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034aa10, 0xc8207457e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7615 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a930, 0xc8205e6180)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7623 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:645 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a930, 0xc8207445c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7613 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc8202970e0)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x490

goroutine 7653 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:645 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034bb90, 0xc820744ea0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7852 [semacquire]:
sync.runtime_Syncsemacquire(0xc8202d4b10)
	/usr/local/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8202d4b00)
	/usr/local/go/src/sync/cond.go:62 +0x9b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Quiesce(0xc82034a930)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:266 +0x1f7
github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).Stop.func1(0xc8205d0140, 0xc820328a20)
	/go/src/github.com/cockroachdb/cockroach/storage/client_test.go:262 +0x2de
created by github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).Stop
	/go/src/github.com/cockroachdb/cockroach/storage/client_test.go:275 +0x6f

goroutine 7619 [chan receive]:
github.com/cockroachdb/cockroach/util.ListenAndServe.func2()
	/go/src/github.com/cockroachdb/cockroach/util/net.go:88 +0x5b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a5b0, 0xc820744540)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7675 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034aa10, 0xc8207457c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7698 [IO wait]:
net.runtime_pollWait(0x7f07b5a7b2d8, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201de760, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201de760, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8201de700, 0xc820930000, 0x1000, 0x1000, 0x0, 0x7f07ba305050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820026858, 0xc820930000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func2(0xc82062ad20, 0x7f07b5a844f8, 0xc820026858, 0x7f07ba31ac58, 0xc8200d1980, 0xc82059bac0, 0xc82095a960)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:210 +0x5d
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:213 +0x90c

goroutine 7672 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034aa10, 0xc820745760)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7622 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:626 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a930, 0xc8207445a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7735 [chan receive]:
github.com/cockroachdb/cockroach/rpc.(*Server).sendResponses(0xc82062afc0, 0x7f07ba31ac58, 0xc8208f2180, 0xc8201927e0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:300 +0xa4
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func1(0xc82062afc0, 0x7f07ba31ac58, 0xc8208f2180, 0xc8201927e0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:207 +0x3f
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:208 +0x89c

goroutine 7704 [chan receive]:
github.com/cockroachdb/cockroach/rpc.(*Server).sendResponses(0xc8203068a0, 0x7f07ba31ac58, 0xc8200d1c80, 0xc82095b5c0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:300 +0xa4
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func1(0xc8203068a0, 0x7f07ba31ac58, 0xc8200d1c80, 0xc82095b5c0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:207 +0x3f
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:208 +0x89c

goroutine 7658 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034bb90, 0xc820744f20)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7661 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc82062a900, 0xece3113e4, 0x2ab36622, 0x2286a40, 0xc820748a80, 0xc82034bb90, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:176 +0x53f
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:219 +0x23f
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034bb90, 0xc820744f80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7620 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1588 +0x8c4
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a930, 0xc820782700)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7617 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a930, 0xc8205e6300)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7705 [IO wait]:
net.runtime_pollWait(0x7f07b5a7ab58, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201ded80, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201ded80, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8201ded20, 0xc82087b000, 0x1000, 0x1000, 0x0, 0x7f07ba305050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200269c0, 0xc82087b000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func2(0xc8203068a0, 0x7f07b5a844f8, 0xc8200269c0, 0x7f07ba31ac58, 0xc8200d1c80, 0xc82059b4e0, 0xc82095b5c0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:210 +0x5d
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:213 +0x90c

goroutine 7699 [IO wait]:
net.runtime_pollWait(0x7f07b5a7b818, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201de450, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201de450, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8201de3f0, 0xc820872000, 0x1000, 0x1000, 0x0, 0x7f07ba305050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820026868, 0xc820872000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/rpc.(*Client).input(0xc82095aa80)
	/usr/local/go/src/net/rpc/client.go:109 +0xbf
created by net/rpc.NewClientWithCodec
	/usr/local/go/src/net/rpc/client.go:201 +0xd2

goroutine 7618 [IO wait]:
net.runtime_pollWait(0x7f07b5a7ae58, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82034b410, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82034b410, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc82034b3b0, 0x0, 0x7f07ba30b4b8, 0xc820581600)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820026580, 0xc820788d20, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a5b0, 0xc820622c90)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7614 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a930, 0xc8205e6080)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7612 [select]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).startStats(0xc820298dc0)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:214 +0xca5
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).(github.com/cockroachdb/cockroach/kv.startStats)-fm()
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:201 +0x20
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a5b0, 0xc82086e4c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7652 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:626 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034bb90, 0xc820744e80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7673 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034aa10, 0xc820745780)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7616 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a930, 0xc8205e61a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7650 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1588 +0x8c4
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034bb90, 0xc820783a80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7633 [chan receive]:
github.com/cockroachdb/cockroach/util.ListenAndServe.func2()
	/go/src/github.com/cockroachdb/cockroach/util/net.go:88 +0x5b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a5b0, 0xc820744e20)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7736 [IO wait]:
net.runtime_pollWait(0x7f07b5a7af18, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201fd250, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201fd250, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8201fd1f0, 0xc820575000, 0x1000, 0x1000, 0x0, 0x7f07ba305050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc82011a260, 0xc820575000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP.func2(0xc82062afc0, 0x7f07b5a844f8, 0xc82011a260, 0x7f07ba31ac58, 0xc8208f2180, 0xc8201d20f0, 0xc8201927e0)
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:210 +0x5d
created by github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP
	/go/src/github.com/cockroachdb/cockroach/rpc/server.go:213 +0x90c

goroutine 7669 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:645 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034aa10, 0xc820745720)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7660 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034bb90, 0xc820744f60)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7706 [IO wait]:
net.runtime_pollWait(0x7f07b5a7b5d8, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201ded10, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201ded10, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8201decb0, 0xc820287000, 0x1000, 0x1000, 0x0, 0x7f07ba305050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200269b0, 0xc820287000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/rpc.(*Client).input(0xc82095b680)
	/usr/local/go/src/net/rpc/client.go:109 +0xbf
created by net/rpc.NewClientWithCodec
	/usr/local/go/src/net/rpc/client.go:201 +0xd2

goroutine 7655 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034bb90, 0xc820744ec0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7666 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1588 +0x8c4
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034aa10, 0xc82086e6a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7674 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034aa10, 0xc8207457a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7651 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:566 +0x119
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034bb90, 0xc820744e60)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7657 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034bb90, 0xc820744f00)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7679 [select]:
github.com/cockroachdb/cockroach/storage.(*idAllocator).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/id_alloc.go:129 +0x911
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a930, 0xc82086ed70)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7656 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034bb90, 0xc820744ee0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7668 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:626 +0x361
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034aa10, 0xc820745700)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7737 [IO wait]:
net.runtime_pollWait(0x7f07b5a7afd8, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201fd100, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201fd100, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8201fd0a0, 0xc8202b9000, 0x1000, 0x1000, 0x0, 0x7f07ba305050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc82011a918, 0xc8202b9000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/rpc.(*Client).input(0xc8201938c0)
	/usr/local/go/src/net/rpc/client.go:109 +0xbf
created by net/rpc.NewClientWithCodec
	/usr/local/go/src/net/rpc/client.go:201 +0xd2

goroutine 7667 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:566 +0x119
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034aa10, 0xc8207456e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7631 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc820296dc0)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x490

goroutine 7611 [select]:
github.com/cockroachdb/cockroach/storage.(*StorePool).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store_pool.go:213 +0x1b6
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a5b0, 0xc820745420)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7659 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034bb90, 0xc820744f40)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7635 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a930, 0xc8205e6380)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7621 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:566 +0x119
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a930, 0xc820744580)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7636 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc8203078c0, 0xece3113e4, 0x2a9bb842, 0x2286a40, 0xc820748a80, 0xc82034a930, 0xc8205b61a0, 0xc820916698)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:176 +0x53f
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1.1(0xc8205b61a0, 0xc8207490b0)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:214 +0x78
github.com/cockroachdb/cockroach/storage.(*storeRangeSet).Visit(0xc820749200, 0xc8202d46c0)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:221 +0x246
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:216 +0x1e1
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a930, 0xc8205e63a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7634 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x329
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a930, 0xc8205e6320)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7632 [IO wait]:
net.runtime_pollWait(0x7f07b5a7aa98, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82034be20, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82034be20, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc82034bdc0, 0x0, 0x7f07ba30b4b8, 0xc8205806c0)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc8200263f0, 0xc820784d20, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a5b0, 0xc820623950)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7664 [IO wait]:
net.runtime_pollWait(0x7f07b5a7b758, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82034ab50, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82034ab50, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc82034aaf0, 0x0, 0x7f07ba30b4b8, 0xc8208eb1c0)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820026450, 0xc8202cad20, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a5b0, 0xc820623f80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7665 [chan receive]:
github.com/cockroachdb/cockroach/util.ListenAndServe.func2()
	/go/src/github.com/cockroachdb/cockroach/util/net.go:88 +0x5b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034a5b0, 0xc820745680)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62

goroutine 7677 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc82062b860, 0xece3113e4, 0x2ac05924, 0x2286a40, 0xc820748a80, 0xc82034aa10, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:176 +0x53f
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:219 +0x23f
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034aa10, 0xc820745800)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x62
FAIL	github.com/cockroachdb/cockroach/storage	39.916s
=== RUN   TestBatchBasics
I0120 05:36:19.240150 656 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I0120 05:36:19.241299 656 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.00s)
=== RUN   TestBatchMerge
I0120 05:36:19.245084 656 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -668,6 +668,7 @@ func TestStoreRangeDownReplicate(t *testing.T) {
 
 	// Expire leader leases one more time, so that any remaining resolutions can
 	// get a leader lease.
+	// TODO(bdarnell): understand why some tests need this.
 	mtc.expireLeaderLeases()
 }
 
@@ -1075,6 +1076,10 @@ func TestRaftAfterRemoveRange(t *testing.T) {
 	// Execute another replica change to ensure that raft has processed
 	// the heartbeat just sent.
 	mtc.replicateRange(roachpb.RangeID(1), 1)
+
+	// Expire leases to ensure any remaining intent resolutions can complete.
+	// TODO(bdarnell): understand why some tests need this.
+	mtc.expireLeaderLeases()
 }
 
 // TestRaftRemoveRace adds and removes a replica repeatedly in an

--- a/storage/replica.go
+++ b/storage/replica.go
@@ -1705,16 +1705,22 @@ func (r *Replica) handleSkippedIntents(intents []intentsWithArg) {
 		// still be careful though, a retry could happen and race with args.
 		args := proto.Clone(item.args).(roachpb.Request)
 		stopper.RunAsyncTask(func() {
+			// Everything here is best effort; give up rather than waiting
+			// too long (helps avoid deadlocks during test shutdown,
+			// although this is imperfect due to the use of an
+			// uninterruptible WaitGroup.Wait in beginCmds).
+			ctxWithDeadline, cancel := context.WithDeadline(ctx, time.Now().Add(rpc.DefaultRPCTimeout))
+			defer cancel()
 			h := roachpb.Header{Timestamp: now}
-			resolveIntents, pErr := r.store.resolveWriteIntentError(ctx, &roachpb.WriteIntentError{
+			resolveIntents, pErr := r.store.resolveWriteIntentError(ctxWithDeadline, &roachpb.WriteIntentError{
 				Intents: item.intents,
 			}, r, args, h, roachpb.PUSH_TOUCH)
 			if wiErr, ok := pErr.GoError().(*roachpb.WriteIntentError); !ok || wiErr == nil || !wiErr.Resolved {
-				log.Warningc(ctx, "failed to push during intent resolution: %s", pErr)
+				log.Warningc(ctxWithDeadline, "failed to push during intent resolution: %s", pErr)
 				return
 			}
-			if pErr := r.resolveIntents(ctx, resolveIntents, true /* wait */, true /* poison */); pErr != nil {
-				log.Warningc(ctx, "failed to resolve intents: %s", pErr)
+			if pErr := r.resolveIntents(ctxWithDeadline, resolveIntents, true /* wait */, true /* poison */); pErr != nil {
+				log.Warningc(ctxWithDeadline, "failed to resolve intents: %s", pErr)
 				return
 			}
 			// We successfully resolved the intents, so we're able to GC from
@@ -1743,8 +1749,6 @@ func (r *Replica) handleSkippedIntents(intents []intentsWithArg) {
 				gcArgs.Keys = append(gcArgs.Keys, roachpb.GCRequest_GCKey{Key: keys.TransactionKey(txn.Key, txn.ID)})
 
 				ba.Add(&gcArgs)
-				ctxWithDeadline, cancel := context.WithDeadline(ctx, time.Now().Add(rpc.DefaultRPCTimeout))
-				defer cancel()
 				if _, pErr := r.addWriteCmd(ctxWithDeadline, ba, nil /* nil */); pErr != nil {
 					log.Warningf("could not GC completed transaction: %s", pErr)
 				}

